### PR TITLE
add undo support.... it works

### PR DESCRIPTION
**Pull request by juliexue**
there are some unused code.... like the timeline..... what should i do with it?

### CODE REVIEW

File: undo_state_runner.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import os
+
```
Comment (davidozhang) [inline undo_state_runner.py:2]: add an extra space. please use a pep-8 linter!

**Review by davidozhang — APPROVED** [review]

**Comment by Nosajool** [comment]
👍 🎉

File: undo_state_runner.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+class UndoStateRunner:
+    strings = {
+        'undo_reflog_command': 'git reset --hard $(git rev-parse --abbrev-ref HEAD)@{${1-1}};'
```
Comment (Nosajool) [inline undo_state_runner.py:6]: I don't think the `${1-1}` does anything in this case

Comment (juliexue) [reply undo_state_runner.py:6]: it works 🖌

**Pull request merged**